### PR TITLE
:electron: Desktop app to work with self signed certificates

### PR DESCRIPTION
**Pull request by MikesGlitch**
#3304

Allowing users to select their self-signed certificate and save it to global-store.json. We manually add the certificate to the trusted cert chain (required for node).

This allows the desktop app to call servers that use self-signed certs. We support .crt and .pem.

test

### CODE REVIEW

File: packages/desktop-client/src/components/manager/ConfigServer.tsx [hunk 1]
```diff
@@ -121,16 +143,42 @@ export function ConfigServer() {
...
+            style={{
+              marginTop: 20,
+              color: theme.errorText,
+              borderRadius: 4,
+              fontSize: 15,
+            }}
+          >
+            {getErrorMessage(error)}
+          </Text>
+          {isElectron() && (
```
Comment (MikesGlitch) [inline packages/desktop-client/src/components/manager/ConfigServer.tsx:157]: Only shown on Electron - browsers don't have the issue. Node has to explicitly trust the cert, that only affects Electron.

File: packages/loot-core/webpack/webpack.browser.config.js [hunk 2]
```diff
@@ -41,6 +41,7 @@ module.exports = {
       process: require.resolve('process/browser'),
       stream: require.resolve('stream-browserify'),
       tls: false,
+      https: false,
```
Comment (MikesGlitch) [inline packages/loot-core/webpack/webpack.browser.config.js:44]: No polyfill required

**Review by MatissJanis — APPROVED** [review]

**Comment by Teprifer** [comment]
@MikesGlitch is there a location for logs for the desktop app? I randomly thought I'd give it a try but it just crashes after selecting my cert.

I'd initially had it installed from the store before, windows auto updated it and that was happening, so uninstalled and nuked the %appdata% dir (nothing in the logs folder), reinstalled, but still crashes unfortunately.

Can't rule out something weird I'd done on this PC at some point though.

**Comment by MikesGlitch** [comment]
Ouch. The logs are lacking - we've been relying on relaying them to devtools. If it's crashing you wont see them.

Can you go to your global-store.json (```%AppData%/Actual/global-store.json```) file and tell me the value of ```server-self-signed-cert```?

I just set my self signed cert up and it's not breaking - I'm wondering what could be different.

**Comment by Teprifer** [comment]
That folder doesn't exist, from other chatter I'd seen I checked `Documents\Actual\` and there were no files until I created a test budget. :(
It only had these three in a folder named after the test budget:
image


I'd previous tried adding the cert(some time ago) in to the cert store so went in and nuked those, but no change.

**Comment by MikesGlitch** [comment]
Yeah, that should be correct, the document folder just contains the budget and nothing else. 

It shouldn't crash after selecting the cert, but it should close and reopen itself (restarting the app).

If that folder doesn't exist that's probably the issue - that's where the app gets its settings. You said you nuked it after uninstalling, are you saying it doesn't recreate if you reinstall? 

Did you uninstall all versions of the app (Windows store and NSIS version) before nuking %AppData%/Actual? They both use that same directory. If you delete that directory with the app already installed, I'd expect the app to crash the app - because that's deleting essential files.

If it's easier, we can continue this on Discord?

**Comment by Teprifer** [comment]
For anyone who finds this in a web search, we worked out it I needed to enter the server name for the server address, not the IP.

Didn't work: https://192.168.1.11:5006
Works: https://blah-server:5006

**Pull request merged**